pages/design_documentation/metadata_contets.md

```diff
@@ -0,0 +1,148 @@
+---
+title: Metadata contents
+last_updated: June 20, 2022
+toc: false
+permalink: metadata_contents.html
+---
+
+OCF metadata comprises multiple sections representing different types of
+metadata. Sections can be assigned to one of few categories determining
+how they are handled:
+
+### Config vs runtime
+
+**Config sections** updates are only done as a part of management operation
+handling. They are always flushed to the cache volume before the management
+operation completes.
+
+**Runtime sections** can be updated at any time, during I/O handling or on
+any asynchronous event like backfill or cleaning. They are flushed only on
+the cache stop/detach. Those sections contain mostly stat counters, which
+are not essential during cache recovery, thus can be safely reset to the
+initial value after dirty shutdown event.
+
+### Fixed size vs variable size
+
+**Fixed size sections** have the same size regardless of the number of cache
+lines. They store cache configuration and statistics.
+
+**Variable size sections** have size that scales linearly with number of cache
+lines. They store information about the cache mapping.
+
+### Flapped vs non-flapped
+
+**Flapped sections** use flapping technique to assure atomic update of metadata
+on the cache volume in case of power failure. They maintain two copies of the
```

Comment:
This is actually not a requirement of this design. This is in fact our use case, but we could be more elastic (e.g. have atomicity per section) and it would still work. So this information should be rather in description of metadata handling flow than in description of particular sections.